File: Java/problem1436.java
class Solution {
    public String destCity(List<List<String>> paths) {
        Map<String, String> map = new HashMap<>();
        for (List<String> path : paths) {
            map.put(path.get(0), path.get(1));
        }
        String result = paths.get(0).get(0);
        while (map.containsKey(result)) {
            result = map.get(result);
        }
        return result;
    }
}

File: python/problem1436.py
class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        start2end = {}
        for path in paths:
            start2end[path[0]] = path[1]
        start = paths[0][0]
        while start in start2end:
            start = start2end[start]
        return start
